corehq/apps/hqwebapp/static/hqwebapp/js/multiselect_utils.js

```diff
@@ -12,18 +12,20 @@ hqDefine('hqwebapp/js/multiselect_utils', [
     var multiselect_utils = {};
 
     var _renderHeader = function (title, action, search) {
-        var header = _.template('<div class="ms-header"><%= headerTitle %><%= actionButton %></div><%= searchInput %>');
-        return header({
-            headerTitle: title,
-            actionButton: action || '',
-            searchInput: search || '',
-        });
+        if (!action){
+            action = '';
+        }
+        if (!search){
+            search = '';
+        }
+        var templateStr = '<div class="ms-header">' + title + action + '</div>' + search;
+        return  _.template(templateStr);
```

Comment:
This change doesn't seem right to me. Similar to my comment on https://github.com/dimagi/commcare-hq/pull/29469#discussion_r608309536, I think templates should not be string-built like this, because that increases the surface area for injection. Instead, my view is that templates should be made from pre-defined static strings that are then rendered.

What was the issue with how it was done before? Also previously this function returned a string, but with your code change here it now returns a _template_.

Reply:
My reading of this is that in this function, `action` and `search` are both safe HTML that has been constructed by other functions using a template. (Well, safe once you include your other changes in this PR, at least.) If that's right then they can be inserted directly using `<%=`, and a comment describing why it's safe. And I think `title` is just a string that should be escape-interpolated with `<%- title %>`. I'm getting this by looking at how it's called, but double-check my work here :)

Reply:
Thanks, @dannyroberts for your suggestions.

> What was the issue with how it was done before?

`action` and `search` both variables had HTML tags. 

```
_.template('<div class="ms-header"><%- headerTitle %><%- actionButton %></div><%- searchInput %>')
```
It would spit out raw HTML on the screen.

> I think templates should not be string-built like this, because that increases the surface area for injection.

The area is not much of my expertise but I am curious to understand why is it so? 

> If that's right then they can be inserted directly using <%=, and a comment describing why it's safe.

That is right, I was thinking of doing the same earlier but I was more focused on getting rid of `<%=` from the codebase so I resorted to concatenating strings to make the template.

I have a question it is related to my previous question, given that we would be okay in using `<%=` if the string is constructed by some other function, other than visually appealing how is different from adding the strings using `+` if we know the strings are safe.

Reply:
For now, I am reverting back the changes and adding a comment on why is safe to do so.

Reply:
Adding a vote for using `<%=` with a comment as to why it's being used, that's clearer than using a different code construct for the sake of avoiding `<%=`. I definitely missed what was going on here when reading through this. The newest code looks good!